### PR TITLE
[ADD] partner_interest_groups

### PR DESCRIPTION
**Pull request by victor-champonnois**
Adds an Interests Groups field (many2many), with configurable categories.
 
I used the terminology "Interests Groups" (cf [wikipedia](https://en.wikipedia.org/wiki/Interest_group_(disambiguation))) because it is flexible enough to accommodate various uses. It's a way to store information about what the partner is interested in.  

For instance, the client for which we develop this module is a federation of structures acting in social economy. A lot of their activities is organizing events, social economy advocacy and communication. They need a structured way to group their partners around what topic they are interested in to improve their outreach.

More technical names could be for instance [Special interest group](https://en.wikipedia.org/wiki/Special_interest_group) or [Advocacy group](https://en.wikipedia.org/wiki/Advocacy_group)

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please don't use plurals in module names as stated in guidelines. `partner_interest_group` is also significant enough, or if you want to highlight that is a multi-valued, you can name it `partner_interest_group_multi` or `partner_interest_group_m2m`.

**Comment by victor-champonnois** [comment]
@pedrobaeza Thanks, I fixed it. 
I cannot run pre-commit because it asks me to sign in to Gitlab, do you know a way around this ?

**Comment by pedrobaeza** [comment]
It seems you don't have rebased your branch over latest changes in origin/16.0

**Comment by victor-champonnois** [comment]
@pedrobaeza thanks, it's done

**Review by flotho — COMMENTED** [review]
Thanks for your support,
Is there any chance to precise what's hiding behind your definition of interest groups.
Even if the code LGTM, I think that some examples or wikipedia references would help people to choose your module.

**Comment by victor-champonnois** [comment]
@flotho Thanks for the feedback. I added an example, alternative terminologies and wikipedia links (see first message).

**Comment by flotho** [comment]
> @flotho Thanks for the feedback. I added an example, alternative terminologies and wikipedia links (see first message).

thanks, any chance to see it in the readme ?

**Comment by victor-champonnois** [comment]
@flotho done !

File: partner_interest_group/models/res_partner_interest_groups.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from odoo import fields, models
+
+
+class ResPartnerInterestGroup(models.Model):
+    _name = "res.partner.interest.groups"
```
Comment (remytms) [inline partner_interest_group/models/res_partner_interest_groups.py]: I suggest not to use plural form here as the object represent one interest group
```python
_name = "res.partner.interest.group"
```

File: partner_interest_group/security/ir.model.access.csv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+"id","name","model_id:id","group_id:id","perm_read","perm_write","perm_create","perm_unlink"
+"access_res_partner_interest_groups_public","res_partner_interest_groups group_public","model_res_partner_interest_groups","base.group_public",1,0,0,0
+"access_res_partner_interest_groups_portal","res_partner_interest_groups group_portal","model_res_partner_interest_groups","base.group_portal",1,0,0,0
+"access_res_partner_interest_groups_partner_manager","res_partner_interest_groups group_partner_manager","model_res_partner_interest_groups","base.group_partner_manager",1,1,1,1
+"access_res_partner_interest_groups_user","res_partner_interest_groups group_user","model_res_partner_interest_groups","base.group_user",1,0,0,0
```
Comment (remytms) [inline partner_interest_group/security/ir.model.access.csv]: Change plural form here too.

File: partner_interest_group/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    interest_groups_ids = fields.Many2many("res.partner.interest.groups")
```
Comment (remytms) [inline partner_interest_group/models/res_partner.py]: Here, I would suggest to remove plural form to `groups` as the plural is indicated by the `ids` write:
```python
interest_group_ids = ...
```

**Review by remytms — CHANGES_REQUESTED** [review]
Please, changes from plural form to singular.

File: partner_interest_group/models/res_partner.py (hunk 3 above)
Comment (victor-champonnois) [reply partner_interest_group/models/res_partner.py]: fixed

File: partner_interest_group/security/ir.model.access.csv (hunk 2 above)
Comment (victor-champonnois) [reply partner_interest_group/security/ir.model.access.csv]: fixed

File: partner_interest_group/models/res_partner_interest_groups.py (hunk 1 above)
Comment (victor-champonnois) [reply partner_interest_group/models/res_partner_interest_groups.py]: fixed

**Comment by victor-champonnois** [comment]
@remytms Fixed :)

File: partner_interest_group/models/res_partner_interest_group.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from odoo import fields, models
+
+
+class ResPartnerInterestGroup(models.Model):
+    _name = "res.partner.interest.group"
+    _description = "Configurable Interest Group for Partners"
+
+    name = fields.Char(string="Interest Group")
+    active = fields.Boolean(default=True)
+    res_partner_id = fields.Many2many("res.partner")
```
Comment (remytms) [inline partner_interest_group/models/res_partner_interest_group.py]: I forgot to mention this in last review, if you have time to change it, it's nice. If not I can live with a `res_partner_id` field.
```suggestion
    partner_id = fields.Many2many("res.partner")
```

**Review by remytms — APPROVED** [review]
A small change, but I approve already.

File: partner_interest_group/models/res_partner_interest_group.py (hunk 4 above)
Comment (victor-champonnois) [reply partner_interest_group/models/res_partner_interest_group.py]: fixed

**Review by flotho — APPROVED** [review]

**Pull request merged**